File: PDOO/Napakalaki-Java/src/NapakalakiGame/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author Usuario
 */
public class Monster {
    
    private String name;
    private int combatLevel;
    private BadConsequence bc;
    private Prize price;
    private int levelChangeAgainstCultistPlayer = 0;
    
    public Monster(String name, int combatLevel, BadConsequence bc, Prize price){
        this.name=name;
        this.combatLevel=combatLevel;
        this.bc=bc;
        this.price=price;
        this.levelChangeAgainstCultistPlayer=0;
    }
    
    public Monster(String n, int cL, BadConsequence bc, Prize p, int IC){
        this.name=n;
        this.combatLevel=cL;
        this.bc=bc;
        this.price=p;
        this.levelChangeAgainstCultistPlayer=IC;
    }

    public String getName() {
        return name;
    }

    public int getCombatLevel() {
        return combatLevel;
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        return (this.combatLevel+this.levelChangeAgainstCultistPlayer);
    }

    public BadConsequence getBc() {
        return bc;
    }

    public Prize getPrice() {
        return price;
    }
    
    @Override
    public String toString(){
        return "Nombre = " + this.getName() + "\nNivel de combate = " + Integer.toString(combatLevel) + 
                "\n " + bc.toString() + "\n" + price.toString();
    }
    
    public int getLevelsGained(){
        return price.getLevel();
    }
    
    public int getTreasuresGained(){
        return price.getTreasures();
    }
}

File: PDOO/Napakalaki-Ruby/lib/Monster.rb
#encoding: utf-8

require_relative 'BadConsequence.rb'
require_relative 'Prize.rb'

module NapakalakiGame
class Monster
    
    attr_reader :name, :combatLevel, :bc, :price, :levelChangeAgainstCultistPlayer
    
    def initialize(name,level,bads,goods,lC)
      @name = name
      @level = level
      @badconsequence = bads
      @prize = goods
      @levelChangeAgainstCultistPlayer = lC
    end
    
    def self.Monster(name,level,bads,goods)
      new(name,level,bads,goods,0)
    end
    
    def self.MonsterC(name,level,bads,goods,lC)
      new(name,level,bads,goods,lC)
    end
    
    def getCombatLevelAgainstCultistPlayer()
      return getCombatLevel + @levelChangeAgainstCultistPlayer
    end
    
    def getName()
      return @name
    end
    def getCombatLevel()
      return @level
    end
    def getBadConsequence()
      return @badconsequence
    end
    
    def getLevelsGained()
        return @prize.levels
    end
    
    def getTreasuresGained()
        return @prize.treasures
    end

    def to_s
        return "Nombre:"+@name.to_s()+" Nivel:"+@level.to_s()+"\n\tMal rollo: "+@badconsequence.to_s()+"\n\tBuen rollo: "+@prize.to_s()
    end
end
end